app/models/semester.rb

```diff
@@ -1,29 +1,32 @@
+# frozen_string_literal: true
+
 class Semester < ActiveRecord::Base
   include Authority::Abilities
   has_many :freeform_shows, dependent: :destroy
   has_many :specialty_shows, dependent: :destroy
   has_many :talk_shows, dependent: :destroy
 
+  before_validation :ensure_beginning_and_ending_are_at_six_am
+
   validates :beginning, :ending, presence: true
-  validate :semesters_dont_overlap
-  validate :beginning_before_ending
+  validate :discrete_semester_dates
```

Comment:
I don't think these actually are the same thing. Let's leave them separate for explicitness.

Reply:
Ooooh okay makes sense. This is fine, then.